lib/importer/factory/object_factory.rb

```diff
@@ -25,11 +25,16 @@ def run
         object
       end
 
+      ## FOR CONSIDERATION: handle a row (i.e. Work) with more than one file:
+      ## currently the file_set is replaced on update
       def update
         raise "Object doesn't exist" unless object
+        @attr = update_attributes
         run_callbacks(:save) do
-          work_actor.update(environment(update_attributes))
+          work_actor.update(environment(@attr))
         end
+        destroy_existing_file_set if object.file_sets.present?
+        attach_file_to_work
```

Comment:
attaching a file to a work is the responsibility of the work actor. Why are you adding it here in duplicate?

Reply:
It is done in the actor stack here: https://github.com/samvera/hyrax/blob/f43e7048bf3961f48812809897b36d84b6577f68/app/actors/hyrax/actors/create_with_files_actor.rb#L10  now.  So you may need to upload the file and then set the uploaded file id in the `:uploaded_files` parameter before calling the actor.

Reply:
Thanks! Will work on this.